src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -853,7 +863,7 @@ public Boolean call() throws InterruptedException {
         final Node node = computer.getNode();
         final String nodeName = node != null ? node.getNodeName() : "unknown";
         if(node != null) {
-            CredentialsProvider.track(node, credentials);
+            CredentialsProvider.track(node, getCredentials());
```

Comment:
Calling a non-final getter in the constructor is generally a bad idea. SSHLauncher is extended in several plugins, and this method is overridden: https://github.com/jenkinsci/docker-plugin/blob/5a2a123cd808a9c82d154cbcc22ef906a6a97907/src/main/java/io/jenkins/docker/connector/DockerComputerSSHConnector.java#L417-L434

Reply:
this is not constructor but `launch` method body.
Maybe you've been confused by github UI to hardly capture the context with a Callable involved